lib/compiler/src/beam_validator.erl

```diff
@@ -2157,6 +2160,49 @@ validate_select_tuple_arity(Fail, [], _, #vst{}=Vst) ->
                    kill_state(SuccVst)
            end).
 
+%%
+%% Validate debug information in `debug_line` instructions.
+%%
+
+validate_debug_line({entry,Args}, Live, #vst{current=St}=Vst) ->
+    case St of
+        #st{numy=none} ->
+            ok;
+        #st{numy=ActualStk} ->
+            error({beam_debug_info,frame_size,none,actual,ActualStk})
+    end,
+    verify_live(Live, Vst),
+    verify_y_init(Vst),
```

Comment:
This rather part is largely shared between the clauses, the only difference I can see is `none` vs `Stk` in the error path. How about breaking this out into a helper, passing `none`/`Stk`?